### PR TITLE
Add NioTransport threads to thread name checks

### PR DESCRIPTION
**Pull request by Tim-Brooks**
We have various assertions that check we never block on transport
threads. This commit adds the thread names for the NioTransport to
these assertions.

With this change I had to fix two places where we were calling blocking
methods from the transport threads.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/transport/Transports.java [hunk 1]
```diff
@@ -29,6 +29,9 @@
     /** threads whose name is prefixed by this string will be considered network threads, even though they aren't */
     public static final String TEST_MOCK_TRANSPORT_THREAD_PREFIX = "__mock_network_thread";
 
+    public static final String NIO_TRANSPORT_WORKER_THREAD_NAME_PREFIX = "nio_transport_worker";
+    public static final String NIO_TRANSPORT_ACCEPTOR_THREAD_NAME_PREFIX = "nio_transport_acceptor";
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/transport/Transports.java]: Should we prefix them with es?

**Review by s1monw — APPROVED** [review]
Left one comment LGTM otherwise

File: core/src/main/java/org/elasticsearch/transport/Transports.java (hunk 1 above)
Comment (tbrooks8) [inline core/src/main/java/org/elasticsearch/transport/Transports.java]: Prefix the constant or the thread name? Or both?

**Review by jasontedor — COMMENTED** [review]
LGTM too.

**Pull request merged**